test/item-action-handler.test.ts

```diff
@@ -0,0 +1,389 @@
+    get: jest.fn(async (id) => items.find(({ id: thisId }) => thisId === id)),
+  } as unknown as ItemService);
+
+// build multiple items url with endpoint (needs to start with '/')
+const buildMultipleItemsUrl = (endpoint, id?) =>
+  `/items${endpoint}${qs.stringify({ id }, { arrayFormat: 'repeat', addQueryPrefix: true })}`;
+
+// dbHandler can be null as we do not use it with the mock itemService
+const dbTransactionHandler = null as unknown as DatabaseTransactionHandler;
+const reply = null as unknown as FastifyReply;
+const log = { debug: () => null } as unknown as FastifyLoggerInstance;
+const item = getDummyItem();
+const itemService = getItemService(item);
+const BUILDER_CLIENT_HOST = CLIENT_HOSTS[0];
+const request = {
+  url: `/items/${item.id}`,
+  method: 'GET',
+  member: GRAASP_ACTOR,
+  params: {},
+  query: {},
+  ip: '',
+  headers: {},
+} as unknown as FastifyRequest;
+
+describe('Create Action Task', () => {
+  it('check geolocation and view properties', async () => {
+    // create a request with valid ip and headers to test view and geolocation
+    const geolocationAndViewRequest = {
+      ...request,
+      ip: '192.158.1.38',
+      headers: {
+        origin: `https://${BUILDER_CLIENT_HOST.hostname}`,
+      },
+    };
+
+    const actions = await itemActionHandler(itemService, {
+      request: geolocationAndViewRequest,
+      reply,
+      log,
+      dbHandler: dbTransactionHandler,
+    });
+    expect(actions[0].geolocation).toBeTruthy();
+    expect(actions[0].view).toEqual(BUILDER_CLIENT_HOST.name);
+  });
+
+  it('returns empty actions array if request does not match any path', async () => {
+    const invalidUrlRequest = {
+      ...request,
+      method: HTTP_METHODS.GET,
+      url: '/hello',
+    };
+    const itemService = getItemService(item);
+
+    jest.spyOn(itemService, 'get');
+    const actions = await itemActionHandler(itemService, {
+      request: invalidUrlRequest,
+      reply,
+      log,
+      dbHandler: dbTransactionHandler,
+    });
+    // actions array should be empty
+    expect(actions).toHaveLength(0);
+  });
+
+  it('GET an item', async () => {
+    const validGetRequest = {
+      ...request,
+      method: HTTP_METHODS.GET,
+      url: `/items/${item.id}`,
+      params: {
+        ...(request.params as object),
+        id: item.id,
+      },
+    };
+
+    const savedActions = await itemActionHandler(itemService, {
+      request: validGetRequest,
+      reply,
+      log,
+      dbHandler: dbTransactionHandler,
+    });
+    expect(savedActions).not.toHaveLength(0);
+    checkActionData(savedActions[0], {
+      itemId: item.id,
+      itemType: item.type,
+      actionType: ACTION_TYPES.GET,
+    });
+  });
+
+  it('GET children', async () => {
+    const validGetChildrenRequest = {
+      ...request,
+      method: HTTP_METHODS.GET,
+      url: `/items/${item.id}/children`,
+      params: {
+        ...(request.params as object),
+        id: item.id,
+      },
+    };
+
```

Comment:
If you're checking a file in particular, you can add this file in `src` next to the corresponding file. Check the build/dist folder doesn't change structure (sometimes the build process includes the test folder while it shouldn't)